### ISSUE TITLE
Tb115 - account template not loaded when changing From address 

### ISSUE THREAD

**Issue by RealRaven2000** [RealRaven2000/SmartTemplates#256]
When selecting a different identity, the new one isn't selected in SmartTemplates 4.1. instead we see the following error log:

```
Uncaught (in promise) TypeError: this.smartTemplate is undefined
    loadIdentity chrome://smarttemplate4/content/smartTemplate-main.js:741
    LoadIdentity chrome://messenger/content/messengercompose/MsgComposeCommands.js:9533
    oncommand chrome://messenger/content/messengercompose/messengercompose.xhtml:1
smartTemplate-main.js:741:11
```

**Comment by RealRaven2000** [comment]
It appears that the new event handler "compose-from-changed" calls `SmartTemplate4.loadIdentity()` with the wrong context.  "this" evaluates to `ChromeWindow chrome://messenger/content/messengercompose/messengercompose.xhtml` and not to the SmartTemplate4 instance of the script. This can be changed by binding a different context in the event listener.

```js
  // old code
  window.addEventListener("compose-from-changed", SmartTemplate4.loadIdentity);
  // new code:
  window.addEventListener("compose-from-changed", SmartTemplate4.loadIdentity.bind(SmartTemplate4));
```

After fixing that, on top of that I am still having problems when detecting the change - because changing the identity during `LoadIdentity()` in core sets the gMsgCompose.bodyModified flag:

![image](https://github.com/RealRaven2000/SmartTemplates/assets/6314674/b5b41283-cf2d-4240-a366-a6a489c1ee0c)

this is a problem because I do not allow changing the mail text if it was already modified (in my mind, this flag is / should be reserved for the _user_ having edited the email)

**Comment by RealRaven2000** [comment]
Ok, here is the bigger picture:

![image](https://github.com/RealRaven2000/SmartTemplates/assets/6314674/47090fbe-66ec-4007-ae09-b0a9430bdfcc)

the problem is exactly when I would need the bodeyModified flag to tell me the truth it pretends that the email was edited. I need to follow the triggering of this and load the identity template but need to know that this event was caused by THunderbird's LoadIdentity and nothing else. I will search the Core database to see whether it is being triggered from anywhere else.

**Comment by RealRaven2000** [comment]
Just playing this through - Thunderbird (I believe falsely) also resets the `bodyModified` flag in this case after the user has done edits. I don't think it should do this, it should store the previous `bodyModfied` state and restore it at the end of `LoadIdentity()`. Ideally it should do that before dispatching the `compose-from-changed` event

**Comment by RealRaven2000** [comment]
Since I cannot determine reliably whether the body was modified when the user changes the identity, I have decided to prompt them:

![image](https://github.com/RealRaven2000/SmartTemplates/assets/6314674/45cccac0-8c9a-481d-bb8c-03c56b9b2dd7)


Clicking OK will discard anything that was written by the user since composer started, Cancel will leave the contents of the email unchanged (It may swap the signature out though).

[smartTemplate-fx-4.1.1pre10.zip](https://github.com/RealRaven2000/SmartTemplates/files/12568314/smartTemplate-fx-4.1.1pre10.zip)

-----

To install the preview version above, please download the zip file and then drag the file into Thunderbird Add-ons Manager. Do not extract file contents, as they won't install.

**Comment by RealRaven2000** [comment]
Implemented in 4.2.1, published 25/09/2023
